src/huggingface_hub/repository.py

```diff
@@ -986,12 +1017,16 @@ def status_method():
                 is_done_method=lambda: process.poll() is not None,
                 status_method=status_method,
                 process=process,
+                post_method=self.lfs_prune if auto_lfs_prune else None,
```

Comment:
Maybe only do this when `blocking=False`? When `blocking=True`, we can run the instruction at the end.

Reply:
This is already the case, this whole block is under `if not blocking`!

https://github.com/huggingface/huggingface_hub/blob/4152f15e7c28fb2afda3f3ece4b5abfd0b8a564e/src/huggingface_hub/repository.py#L1006-L1025

Reply:
Ah sorry, I didn't catch that :-)